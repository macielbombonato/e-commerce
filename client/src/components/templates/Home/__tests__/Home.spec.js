import createTestContext from 'react-cosmos-test/enzyme';

import fixture from '../Home.fixture';

const { mount, getWrapper } = createTestContext({ fixture });

beforeEach(mount);

describe('#Home template', () =>
  it('renders component', () => {
    expect(getWrapper()).toMatchSnapshot();
  }));
