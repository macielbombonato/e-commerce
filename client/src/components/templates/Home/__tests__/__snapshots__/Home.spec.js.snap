// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Home template renders component 1`] = `
<Home
  product={
    Object {
      "list": Array [
        Object {
          "description": "description",
          "id": "123",
          "image": "",
          "name": "product",
          "price": 100,
        },
        Object {
          "description": "description",
          "id": "123",
          "image": "",
          "name": "product",
          "price": 100,
        },
        Object {
          "description": "description",
          "id": "123",
          "image": "",
          "name": "product",
          "price": 100,
        },
        Object {
          "description": "description",
          "id": "123",
          "image": "",
          "name": "product",
          "price": 100,
        },
        Object {
          "description": "description",
          "id": "123",
          "image": "",
          "name": "product",
          "price": 100,
        },
      ],
    }
  }
>
  <Container
    className="mb-4"
    fluid={true}
    tag="div"
  >
    <div
      className="mb-4 container-fluid"
    >
      <Row
        tag="div"
      >
        <div
          className="row"
        >
          <Col
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="col"
            >
              <Slider
                key="slider"
              >
                <Carousel
                  activeIndex={0}
                  interval={5000}
                  keyboard={true}
                  next={[Function]}
                  pause="hover"
                  previous={[Function]}
                  slide={true}
                >
                  <div
                    className="carousel slide"
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    <CarouselIndicators
                      activeIndex={0}
                      items={
                        Array [
                          Object {
                            "altText": "Slide 1",
                            "caption": "",
                            "src": "slide1.jpg",
                          },
                          Object {
                            "altText": "Slide 2",
                            "caption": "",
                            "src": "slide2.png",
                          },
                          Object {
                            "altText": "Slide 3",
                            "caption": "",
                            "src": "slide3.jpg",
                          },
                        ]
                      }
                      onClickHandler={[Function]}
                    >
                      <ol
                        className="carousel-indicators"
                      >
                        <li
                          className="active"
                          key="slide1.jpgSlide 1"
                          onClick={[Function]}
                        />
                        <li
                          className=""
                          key="slide2.pngSlide 2"
                          onClick={[Function]}
                        />
                        <li
                          className=""
                          key="slide3.jpgSlide 3"
                          onClick={[Function]}
                        />
                      </ol>
                    </CarouselIndicators>
                    <div
                      className="carousel-inner"
                      role="listbox"
                    >
                      <CarouselItem
                        altText="Slide 1"
                        appear={false}
                        enter={true}
                        exit={true}
                        in={true}
                        key="slide1.jpg"
                        mountOnEnter={false}
                        onEnter={[Function]}
                        onEntered={[Function]}
                        onEntering={[Function]}
                        onExit={[Function]}
                        onExited={[Function]}
                        onExiting={[Function]}
                        slide={true}
                        src="slide1.jpg"
                        tag="img"
                        timeout={600}
                        unmountOnExit={false}
                      >
                        <Transition
                          appear={false}
                          enter={true}
                          exit={true}
                          in={true}
                          mountOnEnter={false}
                          onEnter={[Function]}
                          onEntered={[Function]}
                          onEntering={[Function]}
                          onExit={[Function]}
                          onExited={[Function]}
                          onExiting={[Function]}
                          timeout={600}
                          unmountOnExit={false}
                        >
                          <div
                            className="carousel-item active"
                          >
                            <img
                              alt="Slide 1"
                              className="d-block img-fluid"
                              src="slide1.jpg"
                            />
                            <CarouselCaption
                              captionHeader=""
                              captionText=""
                            >
                              <div
                                className="carousel-caption d-none d-md-block"
                              >
                                <h3 />
                                <p />
                              </div>
                            </CarouselCaption>
                          </div>
                        </Transition>
                      </CarouselItem>
                      <CarouselItem
                        altText="Slide 2"
                        appear={false}
                        enter={true}
                        exit={true}
                        in={false}
                        key="slide2.png"
                        mountOnEnter={false}
                        onEnter={[Function]}
                        onEntered={[Function]}
                        onEntering={[Function]}
                        onExit={[Function]}
                        onExited={[Function]}
                        onExiting={[Function]}
                        slide={true}
                        src="slide2.png"
                        tag="img"
                        timeout={600}
                        unmountOnExit={false}
                      >
                        <Transition
                          appear={false}
                          enter={true}
                          exit={true}
                          in={false}
                          mountOnEnter={false}
                          onEnter={[Function]}
                          onEntered={[Function]}
                          onEntering={[Function]}
                          onExit={[Function]}
                          onExited={[Function]}
                          onExiting={[Function]}
                          timeout={600}
                          unmountOnExit={false}
                        >
                          <div
                            className="carousel-item"
                          >
                            <img
                              alt="Slide 2"
                              className="d-block img-fluid"
                              src="slide2.png"
                            />
                            <CarouselCaption
                              captionHeader=""
                              captionText=""
                            >
                              <div
                                className="carousel-caption d-none d-md-block"
                              >
                                <h3 />
                                <p />
                              </div>
                            </CarouselCaption>
                          </div>
                        </Transition>
                      </CarouselItem>
                      <CarouselItem
                        altText="Slide 3"
                        appear={false}
                        enter={true}
                        exit={true}
                        in={false}
                        key="slide3.jpg"
                        mountOnEnter={false}
                        onEnter={[Function]}
                        onEntered={[Function]}
                        onEntering={[Function]}
                        onExit={[Function]}
                        onExited={[Function]}
                        onExiting={[Function]}
                        slide={true}
                        src="slide3.jpg"
                        tag="img"
                        timeout={600}
                        unmountOnExit={false}
                      >
                        <Transition
                          appear={false}
                          enter={true}
                          exit={true}
                          in={false}
                          mountOnEnter={false}
                          onEnter={[Function]}
                          onEntered={[Function]}
                          onEntering={[Function]}
                          onExit={[Function]}
                          onExited={[Function]}
                          onExiting={[Function]}
                          timeout={600}
                          unmountOnExit={false}
                        >
                          <div
                            className="carousel-item"
                          >
                            <img
                              alt="Slide 3"
                              className="d-block img-fluid"
                              src="slide3.jpg"
                            />
                            <CarouselCaption
                              captionHeader=""
                              captionText=""
                            >
                              <div
                                className="carousel-caption d-none d-md-block"
                              >
                                <h3 />
                                <p />
                              </div>
                            </CarouselCaption>
                          </div>
                        </Transition>
                      </CarouselItem>
                    </div>
                    <CarouselControl
                      direction="prev"
                      directionText="Previous"
                      onClickHandler={[Function]}
                    >
                      <a
                        className="carousel-control-prev"
                        onClick={[Function]}
                        role="button"
                        tabIndex="0"
                      >
                        <span
                          aria-hidden="true"
                          className="carousel-control-prev-icon"
                        />
                        <span
                          className="sr-only"
                        >
                          Previous
                        </span>
                      </a>
                    </CarouselControl>
                    <CarouselControl
                      direction="next"
                      directionText="Next"
                      onClickHandler={[Function]}
                    >
                      <a
                        className="carousel-control-next"
                        onClick={[Function]}
                        role="button"
                        tabIndex="0"
                      >
                        <span
                          aria-hidden="true"
                          className="carousel-control-next-icon"
                        />
                        <span
                          className="sr-only"
                        >
                          Next
                        </span>
                      </a>
                    </CarouselControl>
                  </div>
                </Carousel>
              </Slider>
            </div>
          </Col>
        </div>
      </Row>
      <Row
        className="mt-3"
        tag="div"
      >
        <div
          className="mt-3 row"
        >
          <Col
            key="0"
            lg="3"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="col col-lg-3"
            >
              <ProductItem
                product={
                  Object {
                    "description": "description",
                    "id": "123",
                    "image": "",
                    "name": "product",
                    "price": 100,
                  }
                }
              >
                <div
                  className="mt-3"
                >
                  <Link
                    replace={false}
                    to="/product/123"
                  >
                    <a
                      href="/product/123"
                      onClick={[Function]}
                    >
                      <Card
                        tag="div"
                      >
                        <div
                          className="card"
                        >
                          <CardImg
                            src="http://localhost:1337/images/"
                            tag="img"
                            width="100%"
                          >
                            <img
                              className="card-img"
                              src="http://localhost:1337/images/"
                              width="100%"
                            />
                          </CardImg>
                          <CardBody
                            tag="div"
                          >
                            <div
                              className="card-body"
                            >
                              <CardTitle
                                tag="h4"
                              >
                                <h4
                                  className="card-title"
                                >
                                  product
                                </h4>
                              </CardTitle>
                              <CardText
                                tag="p"
                              >
                                <p
                                  className="card-text"
                                >
                                  description
                                </p>
                              </CardText>
                              <CardSubtitle
                                tag="h6"
                              >
                                <h6
                                  className="card-subtitle"
                                >
                                  R$ 
                                  100.00
                                </h6>
                              </CardSubtitle>
                            </div>
                          </CardBody>
                        </div>
                      </Card>
                    </a>
                  </Link>
                </div>
              </ProductItem>
            </div>
          </Col>
          <Col
            key="1"
            lg="3"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="col col-lg-3"
            >
              <ProductItem
                product={
                  Object {
                    "description": "description",
                    "id": "123",
                    "image": "",
                    "name": "product",
                    "price": 100,
                  }
                }
              >
                <div
                  className="mt-3"
                >
                  <Link
                    replace={false}
                    to="/product/123"
                  >
                    <a
                      href="/product/123"
                      onClick={[Function]}
                    >
                      <Card
                        tag="div"
                      >
                        <div
                          className="card"
                        >
                          <CardImg
                            src="http://localhost:1337/images/"
                            tag="img"
                            width="100%"
                          >
                            <img
                              className="card-img"
                              src="http://localhost:1337/images/"
                              width="100%"
                            />
                          </CardImg>
                          <CardBody
                            tag="div"
                          >
                            <div
                              className="card-body"
                            >
                              <CardTitle
                                tag="h4"
                              >
                                <h4
                                  className="card-title"
                                >
                                  product
                                </h4>
                              </CardTitle>
                              <CardText
                                tag="p"
                              >
                                <p
                                  className="card-text"
                                >
                                  description
                                </p>
                              </CardText>
                              <CardSubtitle
                                tag="h6"
                              >
                                <h6
                                  className="card-subtitle"
                                >
                                  R$ 
                                  100.00
                                </h6>
                              </CardSubtitle>
                            </div>
                          </CardBody>
                        </div>
                      </Card>
                    </a>
                  </Link>
                </div>
              </ProductItem>
            </div>
          </Col>
          <Col
            key="2"
            lg="3"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="col col-lg-3"
            >
              <ProductItem
                product={
                  Object {
                    "description": "description",
                    "id": "123",
                    "image": "",
                    "name": "product",
                    "price": 100,
                  }
                }
              >
                <div
                  className="mt-3"
                >
                  <Link
                    replace={false}
                    to="/product/123"
                  >
                    <a
                      href="/product/123"
                      onClick={[Function]}
                    >
                      <Card
                        tag="div"
                      >
                        <div
                          className="card"
                        >
                          <CardImg
                            src="http://localhost:1337/images/"
                            tag="img"
                            width="100%"
                          >
                            <img
                              className="card-img"
                              src="http://localhost:1337/images/"
                              width="100%"
                            />
                          </CardImg>
                          <CardBody
                            tag="div"
                          >
                            <div
                              className="card-body"
                            >
                              <CardTitle
                                tag="h4"
                              >
                                <h4
                                  className="card-title"
                                >
                                  product
                                </h4>
                              </CardTitle>
                              <CardText
                                tag="p"
                              >
                                <p
                                  className="card-text"
                                >
                                  description
                                </p>
                              </CardText>
                              <CardSubtitle
                                tag="h6"
                              >
                                <h6
                                  className="card-subtitle"
                                >
                                  R$ 
                                  100.00
                                </h6>
                              </CardSubtitle>
                            </div>
                          </CardBody>
                        </div>
                      </Card>
                    </a>
                  </Link>
                </div>
              </ProductItem>
            </div>
          </Col>
          <Col
            key="3"
            lg="3"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="col col-lg-3"
            >
              <ProductItem
                product={
                  Object {
                    "description": "description",
                    "id": "123",
                    "image": "",
                    "name": "product",
                    "price": 100,
                  }
                }
              >
                <div
                  className="mt-3"
                >
                  <Link
                    replace={false}
                    to="/product/123"
                  >
                    <a
                      href="/product/123"
                      onClick={[Function]}
                    >
                      <Card
                        tag="div"
                      >
                        <div
                          className="card"
                        >
                          <CardImg
                            src="http://localhost:1337/images/"
                            tag="img"
                            width="100%"
                          >
                            <img
                              className="card-img"
                              src="http://localhost:1337/images/"
                              width="100%"
                            />
                          </CardImg>
                          <CardBody
                            tag="div"
                          >
                            <div
                              className="card-body"
                            >
                              <CardTitle
                                tag="h4"
                              >
                                <h4
                                  className="card-title"
                                >
                                  product
                                </h4>
                              </CardTitle>
                              <CardText
                                tag="p"
                              >
                                <p
                                  className="card-text"
                                >
                                  description
                                </p>
                              </CardText>
                              <CardSubtitle
                                tag="h6"
                              >
                                <h6
                                  className="card-subtitle"
                                >
                                  R$ 
                                  100.00
                                </h6>
                              </CardSubtitle>
                            </div>
                          </CardBody>
                        </div>
                      </Card>
                    </a>
                  </Link>
                </div>
              </ProductItem>
            </div>
          </Col>
          <Col
            key="4"
            lg="3"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="col col-lg-3"
            >
              <ProductItem
                product={
                  Object {
                    "description": "description",
                    "id": "123",
                    "image": "",
                    "name": "product",
                    "price": 100,
                  }
                }
              >
                <div
                  className="mt-3"
                >
                  <Link
                    replace={false}
                    to="/product/123"
                  >
                    <a
                      href="/product/123"
                      onClick={[Function]}
                    >
                      <Card
                        tag="div"
                      >
                        <div
                          className="card"
                        >
                          <CardImg
                            src="http://localhost:1337/images/"
                            tag="img"
                            width="100%"
                          >
                            <img
                              className="card-img"
                              src="http://localhost:1337/images/"
                              width="100%"
                            />
                          </CardImg>
                          <CardBody
                            tag="div"
                          >
                            <div
                              className="card-body"
                            >
                              <CardTitle
                                tag="h4"
                              >
                                <h4
                                  className="card-title"
                                >
                                  product
                                </h4>
                              </CardTitle>
                              <CardText
                                tag="p"
                              >
                                <p
                                  className="card-text"
                                >
                                  description
                                </p>
                              </CardText>
                              <CardSubtitle
                                tag="h6"
                              >
                                <h6
                                  className="card-subtitle"
                                >
                                  R$ 
                                  100.00
                                </h6>
                              </CardSubtitle>
                            </div>
                          </CardBody>
                        </div>
                      </Card>
                    </a>
                  </Link>
                </div>
              </ProductItem>
            </div>
          </Col>
        </div>
      </Row>
    </div>
  </Container>
</Home>
`;
